### DISCUSSION TITLE
GPS Pins

### DISCUSSION THREAD

**Discussion by f4dvk** [alanbjohnston/CubeSatSim#341] [General]
Hello,

I would like to know which pins the GPS should be connected to?
What characteristics should it have?

Thank,
Stephane.

**Comment by alanbjohnston** [comment]
Hi Stephane,

A GPS module connects to the Raspberry Pi Pico on the STEM Payload board using connector J12.  This connector has 3.3V, Ground, and Transmit Data (TX2) and Receive Data (RX2).

A serial GPS module such as any of the NEO series will be compatible.  You would wire the 3.3V and Ground to power the module, then connect the Transmit Data (TX) on the GPS board to the Receive Data on the GPS (RX2).

The code running on the Pico uses the Tiny GPS++ Arduino Library (TinyGPS++.h) to decode the NMEA messages to get the latitude, longitude, and elevation.  You can see the code in stempayload/Payload_BME280_MPU6050_XS/Payload_BME280_MPU6050_XS.ino

The only compatibility issue is the bit rate that the GPS module serial port sends the data.  The CubeSatSim code defaults to 9600 bps, which is set by the `Serial2.begin(9600); ` statement.  If your GPS needs 4800 bps or 115200 bps, then you will need to change this.

Other than that, it should work.  If you plug a micro USB cable into the Raspberry Pi Pico, you will see the GPS messages displayed.  Note that most  GPS modules need to be next to a window or outdoors to get a lock, and that lock can take up to 15 minutes, especially the first time, so some patience is needed.

The GPS coordinates are then sent by the Pico to the Pi Zero over the serial port between them.  If the CubeSatSim is in APRS mode, the GPS coordinates will be encoded in each packet.  You can also see the output in the Pi Zero by typing:

`CubeSatSim/config -p`

or by filtering the log output:

`CubeSatSim/log | grep GPS`

Good luck getting it to work!

Alan

**Comment by f4dvk** [comment]
Hi Alan,

Thank you for the description.
The GPS module I have is a GP-02 kit.

Regards,
Stephane.
